### PR TITLE
Updating GC defaults

### PR DESCRIPTION
**Pull request by chflood**
I made the default allocation interval much larger to see if it breaks anything.  I added an env variable to allow changes for testing.

### CODE REVIEW

**Comment by chflood** [comment]
@nanosoldier runbenchmarks(ALL, vs=":master")

**Comment by oscardssmith** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by KristofferC** [comment]
I think Nanosoldier is down at the moment.

**Comment by oscardssmith** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by KristofferC** [comment]
@chflood Note that you need to code quote the runbenchmarks part:

```
`runbenchmarks(ALL, vs=":master")`
```

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/43454#issuecomment-997614133):
```
NanosoldierError: error when preparing/pushing to report repo: failed process: Process(setenv(`git push`; dir="/data/nanosoldier/workdir/NanosoldierReports"), ProcessExited(1)) [1]

```
Unfortunately, the logs could not be uploaded.

**Comment by oscardssmith** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by KristofferC** [comment]
I think the log is too big to u upload so running it again doesn't help. @vtjnash has to manually upload it.

**Comment by vtjnash** [comment]
https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/168a3ef_vs_f0c46b3/report.md

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/43454#issuecomment-998038016):
```
NanosoldierError: error when preparing/pushing to report repo: failed process: Process(setenv(`git push`; dir="/data/nanosoldier/workdir/NanosoldierReports"), ProcessExited(1)) [1]

```
Unfortunately, the logs could not be uploaded.

**Comment by vtjnash** [comment]
https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/3589763_vs_fb3ee73/report.md

**Comment by oscardssmith** [comment]
Do we want to merge this? (asking since we are approaching 1.8 feature freeze).

**Comment by KristofferC** [comment]
This is not really a feature though.

**Comment by chflood** [comment]
OK, so the patch in its current form increases page faults for some of the
nanosoldier benchmarks.   I want to run the benchmarks again with 32mb
instead of 200mb to see if that fixes the problem in the wild like it did
on my laptop.

Christine

**Comment by oscardssmith** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/43454#issuecomment-1005913073):
```
NanosoldierError: error when preparing/pushing to report repo: failed process: Process(setenv(`git push`; dir="/data/nanosoldier/workdir/NanosoldierReports"), ProcessExited(1)) [1]

```
Unfortunately, the logs could not be uploaded.

**Comment by chflood** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by vtjnash** [comment]
https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/c1a287e_vs_1b600f0/report.md

**Comment by chflood** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by vtjnash** [comment]
@nanosoldier `runbenchmarks(!"scalar", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/43454#issuecomment-1009182893) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/27896ce_vs_7b62896/report.md).

**Comment by vtjnash** [comment]
@nanosoldier `runbenchmarks(!"scalar", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/43454#issuecomment-1010436535) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/09c7618_vs_18b825b/report.md).

**Comment by PallHaraldsson** [comment]
Is Nanosoldier reliable for this (and in general)? I was looking at the last two reports, and I did see some unusually large speedup, but also some very bad ratios up to 4.4. E.g. for the last run:

>["array", "comprehension", ("collect", "Vector{Float64}")] | 3.29 (5%)

while in the run before:

>["array", "comprehension", ("collect", "Vector{Float64}")] | 0.30 (5%)

Should we maybe merge this, if there's good reason to believe it will help, even if benchmarks aren't perfect, and we could always undo it before 1.8 release?

**Comment by chflood** [comment]
@nanosoldier `runbenchmarks(!"scalar", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/43454#issuecomment-1024609151) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/b093917_vs_1aac07e/report.md).

File: src/gc.c [hunk 1]
```diff
@@ -3137,12 +3137,12 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
             last_long_collect_interval = gc_num.interval;
         }
         else {
-            // reset interval to default, or at least half of live_bytes
+            // reset interval to half of live_bytes or at most max_collect_interval
             int64_t half = live_bytes/2;
             if (default_collect_interval < half && half <= max_collect_interval)
```
Comment (JeffBezanson) [inline src/gc.c:3142]: Should we also add `if (half < default_collect_interval) gc_num.interval = default_collect_interval;`?

**Comment by chflood** [comment]
Please consider this pull request withdrawn.

**Pull request closed without merging**